src/views/InteractionDesigner.vue

```diff
@@ -33,140 +33,140 @@
-      'isTreeValid',
-      'jsonValidationResults',
-      'validationResults',
-      'builderWidthAdjustment',
-    ]),
-    ...mapState({
-
-      // todo: we'll need to do width as well and use margin-right:365 to allow for sidebar
-      designerWorkspaceHeight: ({trees: {tree, ui}}) => ui.designerWorkspaceHeight,
-      tree: ({trees: {tree, ui}}) => tree,
-      validationResultsEmptyTree: ({trees: {tree, ui}}) => !tree.blocks.length,
-      hasVoice: ({trees: {tree}}) => tree.details.hasVoice,
-      hasSms: ({trees: {tree}}) => tree.details.hasSms,
-      hasUssd: ({trees: {tree}}) => tree.details.hasUssd,
-      hasSocial: ({trees: {tree}}) => tree.details.hasSocial,
-      hasClipboard: ({trees: {tree}}) => tree.details.hasClipboard,
-      blockClasses: ({trees: {ui}}) => ui.blockClasses,
-    }),
-
-    ...mapGetters('flow', ['activeFlow']),
-    ...mapGetters('builder', ['activeBlock', 'isEditable', 'interactionDesignerBoundingClientRect']),
-    ...mapGetters('clipboard', ['isSimulatorActive']),
-
-    jsKey() {
-      return get(this.selectedBlock, 'jsKey')
+  }) readonly appConfig!: object
+  @Prop({
+    type: Object,
+    default() {
+      return {}
     },
+  }) readonly builderConfig!: object
+
+  toolbarHeight = 102
+  // TODO: Move this to BlockClassDetails spec // an inversion can be "legacy types"
+  pureVuejsBlocks = [
+    'CallBackWithCallCenterBlock',
+    'CollaborativeFilteringQuestionBlock',
+    'CollaborativeFilteringRatingBlock',
+    'CollaborativeFilteringRatioBranchBlock',
+    'CreateSubscriberBlock',
+    'CurrentTimeBranchBlock',
+    'DirectorySelectionBlock',
+    'EntitySelectionBlock',
+    'GenerateCodeBlock',
+    'GroupPropertyBlock',
+    'SubscriberPropertiesSnapshotBlock',
+    'SubscriberPropertyBlock',
+    'SummaryBlock',
+    'ValidateCodeBlock',
+    'WebhookBlock',
+    'WebhookContentBlock',
+    'RecordGroupMessageBlock',
+    'PlayGroupMessageBlock',
+  ]
+  simulateClipboard = true
+
+  @Getter isConfigured: any
+  @Getter selectedBlock: any
+  @Getter hasChanges: any
+  @Getter hasIssues: any
+  @Getter isTreeSaving: any
+  @Getter isTreeValid: any
+  @Getter jsonValidationResults: any
+  @Getter validationResults: any
+  @Getter builderWidthAdjustment: any
+
+  // TODO: We'll need to do width as well and use margin-right:365 to allow for sidebar
+  @State(({trees: {tree, ui}}) => ui.designerWorkspaceHeight) designerWorkspaceHeight: any
+  @State(({trees: {tree, ui}}) => tree) tree: any
+  @State(({trees: {tree, ui}}) => !tree.blocks.length) validationResultsEmptyTree: any
+  @State(({trees: {tree}}) => tree.details.hasVoice) hasVoice: any
+  @State(({trees: {tree}}) => tree.details.hasSms) hasSms: any
+  @State(({trees: {tree}}) => tree.details.hasUssd) hasUssd: any
+  @State(({trees: {tree}}) => tree.details.hasSocial) hasSocial: any
+  @State(({trees: {tree}}) => tree.details.hasClipboard) hasClipboard: any
+  @State(({trees: {ui}}) => ui.blockClasses) blockClasses: any
```

Comment:
- [ ] I know the old code used vuex state like this, but `blockClasses` is already declared as getter in https://github.com/FLOIP/flow-builder/blob/feature/VMO-5609/ts-class-components/src/store/trees/trees.js#L104. So, let's use `@Getter blockClasses: any` instead

Reply:
Normally we should have declared these as getters there as well. But I'm ok to leave them here for now.
```
  @State(({trees: {tree}}) => tree.details.hasVoice) hasVoice: any
  @State(({trees: {tree}}) => tree.details.hasSms) hasSms: any
  @State(({trees: {tree}}) => tree.details.hasUssd) hasUssd: any
  @State(({trees: {tree}}) => tree.details.hasSocial) hasSocial: any
  @State(({trees: {tree}}) => tree.details.hasClipboard) hasClipboard: any
```